packages/tailwindcss-language-server/tests/env/v4.test.js

```diff
@@ -666,3 +666,89 @@ defineTest({
     })
   },
 })
+
+defineTest({
+  // This test *always* passes inside Vitest because our custom version of
+  // `Module._resolveFilename` is not called. Our custom implementation is
+  // using enhanced-resolve under the hood which is affected by the `#`
+  // character issue being considered a fragment identifier.
+  //
+  // This most commonly happens when dealing with PNPM packages that point
+  // to a specific commit hash of a git repository.
+  //
+  // To simulate this, we need to:
+  // - Add a local package to package.json
+  // - Symlink that local package to a directory with `#` in the name
+  // - Then run the test in a separate process (`spawn` mode)
+  //
+  // We can't use `file:./a#b` because NPM considers `#` to be a fragment
+  // identifier and will not resolve the path the way we need it to.
+  name: 'v3: require() works when path is resolved to contain a `#`',
+  fs: {
+    'package.json': json`
+      {
+        "dependencies": {
+          "tailwindcss": "3.4.17",
+          "some-pkg": "file:./packages/some-pkg"
+        }
+      }
+    `,
+    'tailwind.config.js': js`
+      module.exports = {
+        presets: [require('some-pkg/config/tailwind.config.js').default]
+      }
+    `,
+    'packages/some-pkg': symlinkTo('packages/some-pkg#c3f1e'),
```

Comment:
Nice helper! Might need that in the main repo soon (there are a bunch of symlink related bug reports). Going to steal this then 😆